### PR TITLE
feat(middlewared/etc): use /etc/cron.d instead appending to system crontab

### PR DESCRIPTION
**Pull request by william-gr**

### CODE REVIEW

**Comment by william-gr** [comment]
Nevermind the review, there is more work to do here

**Comment by themylogin** [comment]
We **do** want to keep all this /etc/cron.daily/man-db, right?

**Comment by william-gr** [comment]
I imagine so, at least for now

File: src/middlewared/middlewared/etc_files/cron.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import shutil
+
+
+def render(service, middleware):
+    shutil.copyfile('/conf/base/etc/crontab.template', '/etc/crontab')
```
Comment (themylogin) [inline src/middlewared/middlewared/etc_files/cron.py]: We can just move this file to template folder and avoid doing this

Comment (william-gr) [reply src/middlewared/middlewared/etc_files/cron.py]: Ah, of course

**Comment by william-gr** [comment]
@themylogin ah, wait, i think my version was the right one. cron(8) says it will reload the files if the mtime of the file has changed. What do you think?

**Comment by themylogin** [comment]
@william-gr when will we need to reload this file, isn't it static?

**Comment by william-gr** [comment]
@themylogin according to the manpage thats the only file that is watched for. So if middlewared cron.d changes alone, it wont reload.

**Comment by themylogin** [comment]
@william-gr have not thought of that. Then the best solution I can see is touching `/etc/crontab` from within `src/middlewared/middlewared/etc_files/crontab` (that way we can keep it being a template, just in case)

**Comment by william-gr** [comment]
I have looked at the source code, and it seems to reload regardless of that, looking for mtime of any cron.d

**Comment by themylogin** [comment]
@william-gr but how does it know when to check if we don't touch /etc/crontab at all?

**Comment by william-gr** [comment]
it only needs to check the first time, which we do. I will check for the mtime of cron.d/middlewared whenever its generated

**Pull request merged**